src/interactivity/base/RemoteConsoleControl.cpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "precomp.h"
+
+#include "RemoteConsoleControl.hpp"
+
+#include "../../inc/HostSignals.hpp"
+
+using namespace Microsoft::Console::Interactivity;
+
+RemoteConsoleControl::RemoteConsoleControl(HANDLE signalPipe) :
+    _pipe{ signalPipe }
+{
+}
+
+#pragma region IConsoleControl Members
+
+template<typename T>
+[[nodiscard]] NTSTATUS _SendTypedPacket(HANDLE pipe, ::Microsoft::Console::HostSignals signalCode, T& payload)
+{
+    // To ensure it's a happy wire format, pack it tight at 1.
+#pragma pack(push, 1)
+    struct HostSignalPacket
+    {
+        ::Microsoft::Console::HostSignals code;
+        T data;
+    };
+#pragma pack(pop)
+
+    HostSignalPacket packet;
+    packet.code = signalCode;
+    packet.data = payload;
+
+    DWORD dwWritten = 0;
+    if (!WriteFile(pipe, &packet, sizeof(packet), &dwWritten, nullptr))

```

Comment:
`RETURN_IF_WIN32_BOOL_FALSE`?

And should `dwWritten` be checked afterwards?
The pipe should be considered broken if a write failed, because if it happened to still work on the next call the receiver code would fail to read the correct amount of `msg.sizeInBytes` and get stuck, being out of sync with the sender logic.

Reply:
unfortunately this set of APIs is returning `NTSTATUS` so my only wil options seem to be the `RETURN_NTSTATUS` and `RETURN_NTSTATUS_MSG` functions without any of the nice flow shorthand. but I've updated to use the wil ones for debuggability and added a test for write length.

Reply:
No on further look... all the wil ones transform...